_posts/2020-10-04-quarkus-native-on-aarch64.adoc

```diff
@@ -0,0 +1,408 @@
+
+#### Keyboard Layout
+
+Find your keymap of choice in the list obtained by issuing the following command:
+
+[source,]
+----
+$ localectl list-keymaps | grep your-locale
+----
+
+And set it into the system config:
+
+
+[source,]
+----
+$ localectl set-keymap _map_
+----
+
+#### Enable internet access through WiFi connection
+
+Get the list of available networks
+
+[source,]
+----
+$ nmcli device wifi list
+----
+
+And connect to the Wi-Fi network of choice
+
+[source,]
+----
+$ nmcli device wifi connect SSID password PASSWORD
+----
+
+#### Disable firewall
+
+Only because it is a demo!!! :-)
+
+[source,]
+----
+$ systemctl disable firewalld
+$ systemctl stop firewalld
+----
+
+#### Create a new user for the device
+
+Thanks to the fedora-arm-installer tool, we've added out public key to the root user. 
+Using root user on a linux system is never a good idea or, let’s say, a good practice.
+For this reason, we’re going to add a new administrative user with wheel privileges to our OS:
+
+[source,]
+----
+$ useradd edge
+$ passwd edge
+$ usermod edge -a -G wheel
+----
+
+
+
+
+
+
+## Connecting the container engine to the quay.io account
+
+If you haven’t got it yet, create a free account on quay.io.
+
+### Sign into Quay.io
+To sign into Quay.io, execute the podman login quay.io command:
+
+Note: If you go to your Quay account settings you can create an encrypted password for more security.
+
+[source,]
+----
+$ podman login quay.io
+Username: myusername
+Password: mypassword
+----
+
+
+
+
+
+## Compiling and deploying your Quarkus native app on Podman
+
+Back on the VM side.
+
+### Building the executable
+
+You’ll need git to download the source code and execute it:
+
+[source,]
+----
+$ sudo dnf install -y git
+----
+
+Clone the quarkus quickstarts repository and pick up the quarkus quickstart of choice.
+
+[source,]
+----
+$ git clone https://github.com/quarkusio/quarkus-quickstarts.git ~/git/quarkus-quickstarts
+----
+
+Enter the root directory of the quickstart of choice (e.g. __getting-started__) and start the build:
+
+[source,]
+----
+$ ./mvnw package -Pnative
+----
+
+The native quarkus executable must be packaged into a container image in order to be able to run it on a container runtime. To achieve this goal, let’s ask the container runtime already installed into your VM to build the container image:
+
+[source,]
+----
+$ sudo podman build -f src/main/docker/Dockerfile.native -t quay.io/$myusername/quarkus-getting-started:1-aarch64 .
+----
+
+Push the container image to the container registry of choice to share it with the edge device:
+
+[source,]
+----
+$ sudo podman push quay.io/$myusername/quarkus-getting-started:1-aarch64
+----
+
+### Deploying the executable
+
+On the RPi:
+
+[source,]
+----
+$ sudo podman run -it --rm -p 8080:8080 --name quarkus-getting-started quay.io/$myusername/quarkus-getting-started:1-aarch64
+----
+
+
```

Comment:
i sugguest you put some summary/conclusion here - maybe a screenshot of the running app or something ;)

Reply:
Sorry, I completely forgot about that. I'm happy to add conclusions, logs that certify the startup time and a screenshot of the memory footprint